### PR TITLE
new logging: cleaned up api/v1/core.py

### PR DESCRIPTION
**Pull request by p0psicles**
* Replaced double quotes with single, where possible.
* Removed ex.
* Replaced %s string parsing.
* Removed string concatenations with formats.

- [ ] PR is based on the DEVELOP branch
- [ ] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [ ] Read the [contribution guide](https://github.com/pymedusa/Medusa/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

**Comment by p0psicles** [comment]
Sorry for the big PR. I've also replaced the quotes while I was at it.
You'll have to do some searching to really review the logging changes.

File: medusa/server/api/v1/core.py [hunk 1]
```diff
@@ -61,7 +62,10 @@
 from six import iteritems, text_type
 from tornado.web import RequestHandler
 
-indexer_ids = ["indexerid", "tvdbid", "tvmazeid", "tmdbid"]
+log = BraceAdapter(logging.getLogger(__name__))
+log.logger.addHandler(logging.NullHandler())
+
+indexer_ids = ['indexerid', 'tvdbid', 'tvmazeid', 'tmdbid']
```
Comment (medariox) [inline medusa/server/api/v1/core.py]: Maybe capitalize to make clear this is a constant.

File: medusa/server/api/v1/core.py [hunk 2]
```diff
@@ -116,19 +120,19 @@ def get(self, *args, **kwargs):
...
 
             _call_dispatcher = profile(_call_dispatcher, immediate=True)
-            del kwargs["profile"]
+            del kwargs['profile']
 
         try:
             out_dict = _call_dispatcher(args, kwargs)
         except Exception as e:  # real internal error oohhh nooo :(
-            logger.log(u"API :: " + ex(e), logger.ERROR)
+            log.exception(u'API :: {0}', e)
```
Comment (medariox) [inline medusa/server/api/v1/core.py]: {0!r} instead?

File: medusa/server/api/v1/core.py [hunk 3]
```diff
@@ -145,16 +149,17 @@ def _out_as_image(self, _dict):
...
             if callback:
-                out = callback + '(' + out + ');'  # wrap with JSONP call if requested
-        except Exception as e:  # if we fail to generate the output fake an error
-            logger.log(u"API :: " + traceback.format_exc(), logger.DEBUG)
-            out = '{"result": "%s", "message": "error while composing output: %s"}' % \
-                  (result_type_map[RESULT_ERROR], ex(e))
+                out = '{0}({1});'.format(callback, out)  # wrap with JSONP call if requested
+        except Exception as error:  # if we fail to generate the output fake an error
+            log.exception(u'API :: Traceback')
+            out = '{{"result": "{0}", "message": "error while composing output: {1!r}"}}'.format(
```
Comment (medariox) [inline medusa/server/api/v1/core.py]: Shouldn't this be a dict?

File: medusa/server/api/v1/core.py [hunk 4]
```diff
@@ -163,25 +168,25 @@ def call_dispatcher(self, args, kwargs):  # pylint:disable=too-many-branches
...
             for cmd in commands:
                 cur_args, cur_kwargs = self.filter_params(cmd, args, kwargs)
 
-                if len(cmd.split("_")) > 1:
-                    cmd, cmd_index = cmd.split("_")
+                if len(cmd.split('_')) > 1:
+                    cmd, cmd_index = cmd.split('_')
 
-                logger.log(u"API :: " + cmd + ": cur_kwargs " + str(cur_kwargs), logger.DEBUG)
+                log.debug(u'API :: {0}: cur_kwargs', cmd, cur_kwargs)
```
Comment (medariox) [inline medusa/server/api/v1/core.py]: Missing {1} ?

File: medusa/server/api/v1/core.py [hunk 5]
```diff
@@ -281,36 +286,36 @@ def return_help(self):
...
+            msg = self._help['desc']
         return _responds(RESULT_SUCCESS, self._help, msg)
 
     def return_missing(self):
         if len(self._missing) == 1:
-            msg = "The required parameter: '" + self._missing[0] + "' was not set"
+            msg = 'The required parameter: {0!r} was not set'.format(self._missing[0])
         else:
-            msg = "The required parameters: '" + "','".join(self._missing) + "' where not set"
+            msg = 'The required parameters: {0!r} where not set"'.format("','".join(self._missing))
```
Comment (medariox) [inline medusa/server/api/v1/core.py]: Something went wrong here.

File: medusa/server/api/v1/core.py (hunk 3 above)
Comment (p0psicles) [inline medusa/server/api/v1/core.py]: It's parsed into json. I'm rewriting using. Format now. Haven't looked into rewriting it to parse from dict.

File: medusa/server/api/v1/core.py (hunk 4 above)
Comment (p0psicles) [inline medusa/server/api/v1/core.py]: Yes good one

File: medusa/server/api/v1/core.py (hunk 2 above)
Comment (p0psicles) [inline medusa/server/api/v1/core.py]: Yes

File: medusa/server/api/v1/core.py (hunk 5 above)
Comment (p0psicles) [inline medusa/server/api/v1/core.py]: I think only the double quote at the end shouldn't be there. I'll do some tests.

Comment (labrys) [inline medusa/server/api/v1/core.py]: Also "were not set" not "where not set"

File: medusa/server/api/v1/core.py (hunk 1 above)
Comment (labrys) [inline medusa/server/api/v1/core.py]: If its a constant it should be a tuple.

**Comment by fernandog** [comment]
so no one will review&approve this PR @p0psicles took a lot of effort to make?
tons of log PR were merged in this is stalled?

File: medusa/server/api/v1/core.py (hunk 3 above)
Comment (medariox) [inline medusa/server/api/v1/core.py]: What I mean is:
_Before:_
```
'{"result": "%s", "message": "error while composing output: %s"}'
```
_After_
```
'{{"result": "{0}", "message": "error while composing output: {1!r}"}}'
```
Why the additional curly braces?

File: medusa/server/api/v1/core.py [hunk 6]
```diff
@@ -163,25 +168,25 @@ def call_dispatcher(self, args, kwargs):  # pylint:disable=too-many-branches
             or calls the TVDBShorthandWrapper when the first args element is a number
             or returns an error that there is no such cmd
         """
-        logger.log(u"API :: all args: '" + str(args) + "'", logger.DEBUG)
-        logger.log(u"API :: all kwargs: '" + str(kwargs) + "'", logger.DEBUG)
+        log.debug(u'API :: all args: {0!r}', args)
+        log.debug(u'API :: all kwargs: {0!r}', kwargs)
```
Comment (medariox) [inline medusa/server/api/v1/core.py]: Shouldn't this stay str instead of repr?

Comment (p0psicles) [inline medusa/server/api/v1/core.py]: It will place it between single quotes.

Comment (medariox) [inline medusa/server/api/v1/core.py]: Okay. I still think repr should be used for object representation. If we really need to have the single quotes (do we need them anyway?), we can use double quotes for the outer quotes.

Comment (p0psicles) [inline medusa/server/api/v1/core.py]: I kind of followed labrys lead here. But I'm fine with either way.

File: medusa/server/api/v1/core.py [hunk 7]
```diff
@@ -192,11 +197,11 @@ def call_dispatcher(self, args, kwargs):  # pylint:disable=too-many-branches
                         elif _is_int(cmd):
                             cur_out_dict = TVDBShorthandWrapper(cur_args, cur_kwargs, cmd).run()
                         else:
-                            cur_out_dict = _responds(RESULT_ERROR, "No such cmd: '" + cmd + "'")
+                            cur_out_dict = _responds(RESULT_ERROR, 'No such cmd: {0!r}'.format(cmd))
                     except ApiError as error:  # Api errors that we raised, they are harmless
-                        cur_out_dict = _responds(RESULT_ERROR, msg=ex(error))
+                        cur_out_dict = _responds(RESULT_ERROR, msg=error)
```
Comment (medariox) [inline medusa/server/api/v1/core.py]: msg expects a string, so maybe ```msg=error.message```?

File: medusa/server/api/v1/core.py [hunk 8]
```diff
@@ -192,11 +197,11 @@ def call_dispatcher(self, args, kwargs):  # pylint:disable=too-many-branches
...
                 else:  # if someone chained one of the forbidden commands they will get an error for this one cmd
-                    cur_out_dict = _responds(RESULT_ERROR, msg="The cmd '" + cmd + "' is not supported while chaining")
+                    cur_out_dict = _responds(RESULT_ERROR, msg='The cmd {0!r} is not supported while chaining'.fomrat(cmd))
```
Comment (medariox) [inline medusa/server/api/v1/core.py]: ```format``` instead of ```fomrat``` (I blame rato for this one)

Comment (labrys) [inline medusa/server/api/v1/core.py]: 'The cmd `fomrat` is not supported while chaining'
Or any other time really

Comment (medariox) [inline medusa/server/api/v1/core.py]: I think this is the only critical change needed. After that we can start testing it. @p0psicles?

Comment (p0psicles) [inline medusa/server/api/v1/core.py]: Next on my list

File: medusa/server/api/v1/core.py (hunk 3 above)
Comment (p0psicles) [inline medusa/server/api/v1/core.py]: Because I'm using format now instead of %s. Format uses {} as special chars. So you have to escape them.

**Comment by p0psicles** [comment]
Anything that I still need to look at? @medariox @labrys ?

**Pull request merged**